File: java/UMLGraph/src/org/daisy/urakawa/PropertyFactory.java
package org.daisy.urakawa;

import org.daisy.urakawa.property.GenericPropertyFactory;
import org.daisy.urakawa.property.channel.ChannelsPropertyFactory;
import org.daisy.urakawa.property.xml.XmlPropertyFactory;

/**
 * <p>
 * This is the factory that creates all types of built-in
 * {@link org.daisy.urakawa.property.Property} instances.
 * </p>
 * <p>
 * Note: this interface assembles a set of other interfaces, but does not
 * introduce new methods itself.
 * </p>
 * 
 * @depend - Aggregation 1 org.daisy.urakawa.Presentation
 * @depend - Create - org.daisy.urakawa.core.property.Property
 * @depend - Create - org.daisy.urakawa.property.xml.XmlAttribute
 * @depend - Create - org.daisy.urakawa.property.xml.XmlProperty
 * @depend - Create - org.daisy.urakawa.property.channel.ChannelsProperty
 * @leafInterface see {@link org.daisy.urakawa.LeafInterface}
 * @see org.daisy.urakawa.LeafInterface
 * @stereotype OptionalLeafInterface
 */
public interface PropertyFactory extends WithPresentation,
		GenericPropertyFactory, XmlPropertyFactory, ChannelsPropertyFactory {
}

File: branches/new-media-factory-design-try-1/csharp/core/property/PropertyFactory.cs
using System;
using System.Xml;
using urakawa.core;
using urakawa.property.channel;
using urakawa.xuk;

namespace urakawa.property
{
    /// <summary>
    /// Factory for creating <see cref="Property"/>s and derived types
    /// </summary>
    public sealed class PropertyFactory : GenericFactory<Property>
    {

        /// <summary>
        /// Creates a <see cref="ChannelsProperty"/>
        /// </summary>
        /// <returns>The created <see cref="ChannelsProperty"/></returns>
        public ChannelsProperty CreateChannelsProperty()
        {
            return Create<ChannelsProperty>();
        }

        /// <summary>
        /// Creates an <see cref="urakawa.property.xml.XmlProperty"/> instance
        /// </summary>
        /// <returns>The created instance</returns>
        public urakawa.property.xml.XmlProperty CreateXmlProperty()
        {
            return Create<urakawa.property.xml.XmlProperty>();
        }
    }
}